pkg/operator/operator.go

```diff
@@ -86,6 +89,19 @@ type Operator struct {
 }
 
 func NewOperator(ctx context.Context, operator *operator.Operator) (context.Context, *Operator) {
+	kubeletCompatibilityAnnotationKey := fmt.Sprintf("%s/%s", apis.CompatibilityGroup, "v1beta1-kubelet-conversion")
+	// we are going to panic if any of the customer nodepools contain
+	// compatibility.karpenter.sh/v1beta1-kubelet-conversion
+	restConfig := clinetconfig.GetConfigOrDie()
+	kubeClient := lo.Must(client.New(restConfig, client.Options{}))
+	nodePoolList := &karpv1.NodePoolList{}
+	lo.Must0(kubeClient.List(ctx, nodePoolList))
+	for _, np := range nodePoolList.Items {
+		if _, found := np.Annotations[kubeletCompatibilityAnnotationKey]; found {
+			panic(fmt.Sprintf("expected to %s nodepool to have %s removed", np.Name, kubeletCompatibilityAnnotationKey))
+		}
+	}
```

Comment:
nit: for consistency, let's use `log.Fatalf`. Also, it would be nice to print out all of the effected NodePools.
```suggestion
    npNames := lo.FilterMap(nodePoolList.Items, func(np karpv1.NodePool, _ int) (string, bool) {
        _, ok := np.Annotations[kubeletCompatibilityAnnotationKey]
        return np.Name, ok
    })
    if len(npNames) != 0 {
        log.Fatalf("The kubelet compatiblity annotation, %s, is not supported on Karpenter v1.1+. Please refer to the upgrade guide in the docs. The following NodePools still have the compatiblity annotation: %s", kubeletCompatiblityAnnotation, strings.Join(npNames, ", ")
    }
```